wakuv2/common/filter.go

```diff
@@ -124,16 +139,15 @@ func (fs *Filters) AllTopics() []TopicType {
 // If the filter's Topics array is empty, it will be tried on every topic.
 // Otherwise, it will be tried on the topics specified.
 func (fs *Filters) addTopicMatcher(watcher *Filter) {
-	if len(watcher.Topics) == 0 && (watcher.PubsubTopic == relay.DefaultWakuTopic || watcher.PubsubTopic == "") {
+	if len(watcher.ContentTopics) == 0 && (watcher.PubsubTopic == relay.DefaultWakuTopic || watcher.PubsubTopic == "") {
 		fs.allTopicsMatcher[watcher] = struct{}{}
 	} else {
 		filtersPerContentTopic, ok := fs.topicMatcher[watcher.PubsubTopic]
 		if !ok {
 			filtersPerContentTopic = make(ContentTopicToFilter)
 		}
 
-		for _, t := range watcher.Topics {
-			topic := BytesToTopic(t)
+		for topic := range watcher.ContentTopics {
```

Comment:
I'm curious about this change. Wouldnt `topic` be an integer here?

Reply:
`ContentTopics` is now a map, so `topic` iterates through the map's keys.